content_api/publish/service.py

```diff
@@ -38,10 +38,18 @@ def publish(self, item, subscribers=[]):
 
         :param item: item to publish
         """
-
         if not self._filter_item(item):
-            doc = self.formatter._transform_to_ninjs(item, self.subscriber)
             now = utcnow()
+            try:
+                # do we already have an item id db?
+                stored_item = self.find_one(req=None, _id=item['_id'])
```

Comment:
would rather put this into _create_doc which is already checking previous version so it's not fetched twice